projects/scion/toolkit.internal/widgets/src/tabbar/tabbar.component.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2018-2019 Swiss Federal Railways
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ *  SPDX-License-Identifier: EPL-2.0
+ */
+
+import { AfterContentInit, ChangeDetectionStrategy, Component, ContentChildren, QueryList, ViewChild, ViewContainerRef } from '@angular/core';
+import { SciTabDirective } from './tab.directive';
+import { startWith, tap } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+
+/**
+ * Organizes content into separate tabs where only one tab can be visible at a time.
+ *
+ * A tab's content is constructed lazily when displayed for the first time. Tab content is not destroyed when selecting another tab.
+ *
+ * Tabs are modelled as content children in the form of a `<ng-template>` decorated with the `sciTab` directive, as following:
+ *
+ * ```
+ * <sci-tabbar>
+ *   <ng-template sciTab [label]="item.label" *ngFor="let item of items$ | async">
+ *     ...
+ *   </ng-template>
+ * </sci-tabbar>
+ * ```
+ */
+@Component({
+  selector: 'sci-tabbar',
+  templateUrl: './tabbar.component.html',
+  styleUrls: ['./tabbar.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SciTabbarComponent implements AfterContentInit {
+
+  @ViewChild('tabcontent', {read: ViewContainerRef, static: true})
+  private _vcr: ViewContainerRef;
+  private _activeTab: SciTabDirective;
+
+  /** @internal */
+  @ContentChildren(SciTabDirective)
+  public tabs: QueryList<SciTabDirective>;
+
+  /** @internal */
+  public tabs$: Observable<QueryList<SciTabDirective>>;
+
+  public ngAfterContentInit(): void {
+    this.tabs$ = this.tabs.changes
+      .pipe(
+        startWith(this.tabs),
+        tap(tabs => this.onTabsChange(tabs))
+      );
+  }
+
+  private onTabsChange(tabs: QueryList<SciTabDirective>): void {
+    // Select the first tab if no tab is selected or if the active tab was removed
+    if (!this._activeTab || !tabs.find(it => it === this._activeTab)) {
+      this.selectTab(tabs.first);
+    }
+  }
+
+  public onTabSelect(tab: SciTabDirective | undefined) {
+    this.selectTab(tab);
+  }
+
+  private selectTab(tab: SciTabDirective | undefined) {
+    // Deactivate the currently selected tab, if any.
+    const selectedTab = this.tabs.find(it => it.isContentAttached());
+    if (selectedTab) {
```

Comment:
Do nothing when clicking on the already active tab, as follows:

```
const selectedTab = this.tabs.find(it => it.isContentAttached());
if (selectedTab && selectedTab === tab) {
   return;
}
```

Reply:
done